third_party/s2n-bignum/include/s2n-bignum_aws-lc.h

```diff
@@ -17,42 +17,69 @@
 // C prototypes for s2n-bignum functions used in AWS-LC
 // ----------------------------------------------------------------------------
 
+// For some functions there are additional variants with names ending in
+// "_alt". These have the same core mathematical functionality as their
+// non-"alt" versions, but can be better suited to some microarchitectures:
+//
+//      - On x86, the "_alt" forms avoid BMI and ADX instruction set
```

Comment:
Does this actually matter? I assume the set of "alt" function is not enough to implement all the feature we want on top of the p384 curve.

Reply:
Not sure I understand. I'll give an example of what's going on. s2n-bignum has `bignum_add_p384` function that is generic in a sense that it doesn't use any special x86 instruction. There is also `bignum_mul_p384` which uses bmi2 and adx instructions that are not available on all x86 CPUs, and the alternative version `bignum_mul_p384_alt` that is generic. So if aws-lc is run on a:
1. CPU with bmi2 and adx then we use `bignum_add_p384` and `bignum_mul_p384`,
2. CPU without bmi2 and adx then we use `bignum_add_p384` and `bignum_mul_p384_alt`.

Reply:
a'ha - I recall now!
Thanks for explaining it on the 5 years old level, needed that :)